forms/NumericField.php

```diff
@@ -30,7 +30,7 @@ public function setValue($value, $data = array()) {
 		// If passing in a non-string number, or a value
 		// directly from a DataObject then localise this number
 
-		if(is_int($value) || is_float($value) || $data instanceof DataObject) {
+		if(is_numeric($value) || is_int($value) || is_float($value) || $data instanceof DataObject) {
```

Comment:
The way it was is intentional; A numeric string should be parsed in the current locale. the is_int case is to explicitly capture REAL integers that do not not need to be parsed.